opm/simulators/wells/StandardWell_impl.hpp

```diff
@@ -2263,6 +2240,91 @@ namespace Opm
 
 
 
+    template<typename TypeTag>
+    void
+    StandardWell<TypeTag>::
+    updateProductivityIndex(const Simulator& ebosSimulator,
+                            const WellProdIndexCalculator& wellPICalc,
+                            WellState& well_state,
+                            DeferredLogger& deferred_logger) const
+    {
+        auto fluidState = [&ebosSimulator, this](const int perf)
+        {
+            const auto cell_idx = this->well_cells_[perf];
+            return ebosSimulator.model()
+               .cachedIntensiveQuantities(cell_idx, /*timeIdx=*/ 0)->fluidState();
+        };
+
+        auto isInjecting = [this](const typename StandardWell<TypeTag>::FluidState& fs,
+                                  const int perf) -> bool
+        {
+            const auto cell_press = this->extendEval(this->getPerfCellPressure(fs));
```

Comment:
Suggest using `getPerfCellPressure(fs).value()` to do this in doubles instead of dragging around AD objects needlessly here.

Reply:
Okay, I'll do that.